File: src/utils/output.go
package utils

import (
	"fmt"
	"os"
	"strings"

	"github.com/common-nighthawk/go-figure"
	"github.com/fatih/color"
	"github.com/olekukonko/tablewriter"
)

const (
	SOFTWARE_VERSION = "2.0.0"
)

// ShowLogo displays the STARLABS ASCII art logo
func ShowLogo() {
	// Create ASCII art with a different font
	myFigure := figure.NewFigure("STAR LABS", "standard", true)
	logo := myFigure.String()

	// Create color objects for blue gradient effect
	blue := color.New(color.FgHiBlue, color.Bold)

	// Split logo into lines
	lines := strings.Split(logo, "\n")

	fmt.Println() // Empty line before logo

	// Print each line in bold blue
	for _, line := range lines {
		if len(strings.TrimSpace(line)) > 0 {
			blue.Println(line)
		}
	}
	fmt.Println() // Empty line after logo
}

// ShowDevInfo displays development and version information
func ShowDevInfo() {
	// Create blue color
	blue := color.New(color.FgHiBlue)

	// Box drawing characters
	const (
		topLeft     = "╔"
		topRight    = "╗"
		bottomLeft  = "╚"
		bottomRight = "╝"
		horizontal  = "═"
		vertical    = "║"
		separator   = "----------------------------------------"
	)

	// Create the info box content
	width := 42
	horizontalLine := strings.Repeat(horizontal, width-2)

	info := []string{
		topLeft + horizontalLine + topRight,
		vertical + centerText("Twitter Bot "+SOFTWARE_VERSION, width-2) + vertical,
		vertical + separator + vertical,
		vertical + centerText("", width-2) + vertical,
		vertical + centerText("GitHub: https://github.com/StarLabs", width-2) + vertical,
		vertical + centerText("", width-2) + vertical,
		vertical + centerText("Developer: https://t.me/StarLabsTech", width-2) + vertical,
		vertical + centerText("Chat: https://t.me/StarLabsChat", width-2) + vertical,
		vertical + centerText("", width-2) + vertical,
		bottomLeft + horizontalLine + bottomRight,
	}

	fmt.Println() // Empty line before info box

	// Print all lines in the same blue color
	for _, line := range info {
		blue.Println(line)
	}
	fmt.Println() // Empty line after info box
}

// centerText centers the text within the given width
func centerText(text string, width int) string {
	padding := width - len(text)
	if padding <= 0 {
		return text
	}
	leftPad := padding / 2
	rightPad := padding - leftPad
	return strings.Repeat(" ", leftPad) + text + strings.Repeat(" ", rightPad)
}

// PrintStatistics prints a beautiful table with processing statistics
func PrintStatistics(total, processed, success, failed, locked, suspended int) {
	// Create new table
	table := tablewriter.NewWriter(os.Stdout)

	// Configure table style
	table.SetHeader([]string{"Statistics", "Value"})
	table.SetBorder(true)
	table.SetAlignment(tablewriter.ALIGN_LEFT)
	table.SetHeaderAlignment(tablewriter.ALIGN_CENTER)

	// Make table bigger
	table.SetColumnSeparator("│")
	table.SetRowSeparator("─")
	table.SetCenterSeparator("┼")
	table.SetHeaderLine(true)
	table.SetColWidth(30)
	table.SetAutoWrapText(false)

	// Set header colors (cyan)
	table.SetHeaderColor(
		tablewriter.Colors{tablewriter.FgHiCyanColor, tablewriter.Bold},
		tablewriter.Colors{tablewriter.FgHiCyanColor, tablewriter.Bold},
	)

	// Set different colors for each row
	table.SetColumnColor(
		tablewriter.Colors{tablewriter.FgHiWhiteColor},
		tablewriter.Colors{}, // Will be set per row
	)

	// Add data rows with different colors for values
	data := []struct {
		label string
		value int
		color int
	}{
		{"Total", total, tablewriter.FgHiWhiteColor},
		{"Processed", processed, tablewriter.FgHiWhiteColor},
		{"Success", success, tablewriter.FgHiGreenColor},
		{"Failed", failed, tablewriter.FgHiYellowColor},
		{"Locked", locked, tablewriter.FgHiRedColor},
		{"Suspended", suspended, tablewriter.FgHiRedColor},
	}

	for _, row := range data {
		table.Rich(
			[]string{row.label, fmt.Sprintf("%d", row.value)},
			[]tablewriter.Colors{
				{tablewriter.FgHiWhiteColor},
				{row.color},
			},
		)
	}

	fmt.Println()
	table.Render()
	fmt.Println()
}

func printStatRow(borderColor *color.Color, valueColor *color.Color, label string, value int, vertical string) {
	borderColor.Printf(vertical)
	fmt.Printf(" %-16s ", label)
	borderColor.Printf(vertical)
	valueColor.Printf(" %8d ", value)
	borderColor.Println(vertical)
}

File: src/utils/output.py
import os
from typing import List
from rich.console import Console
from rich.text import Text
from tabulate import tabulate
from rich.table import Table
from rich import box


def show_logo():
    """Отображает стильный логотип STARLABS"""
    # Очищаем экран
    os.system("cls" if os.name == "nt" else "clear")

    console = Console()

    # Создаем звездное небо со стилизованным логотипом
    logo_text = """
✦ ˚ . ⋆   ˚ ✦  ˚  ✦  . ⋆ ˚   ✦  . ⋆ ˚   ✦ ˚ . ⋆   ˚ ✦  ˚  ✦  . ⋆   ˚ ✦  ˚  ✦  . ⋆ ✦ ˚ 
. ⋆ ˚ ✧  . ⋆ ˚  ✦ ˚ . ⋆  ˚ ✦ . ⋆ ˚  ✦ ˚ . ⋆  ˚ ✦ . ⋆ ˚  ✦ ˚ . ⋆  ˚ ✦ . ⋆  ˚ ✦ .✦ ˚ . 
·˚ ⋆｡⋆｡. ★ ·˚ ★ ·˚ ⋆｡⋆｡. ★ ·˚ ★ ·˚ ⋆｡⋆｡. ★ ·˚ ★ ·˚ ⋆｡⋆｡. ★ ·˚ ⋆｡⋆｡. ★ ·˚ ★ ·˚ ·˚ ★ ·˚
✧ ⋆｡˚✦ ⋆｡  ███████╗████████╗ █████╗ ██████╗ ██╗      █████╗ ██████╗ ███████╗  ⋆｡ ✦˚⋆｡ 
★ ·˚ ⋆｡˚   ██╔════╝╚══██╔══╝██╔══██╗██╔══██╗██║     ██╔══██╗██╔══██╗██╔════╝  ✦˚⋆｡ ˚· 
⋆｡✧ ⋆ ★    ███████╗   ██║   ███████║██████╔╝██║     ███████║██████╔╝███████╗   ˚· ★ ⋆
˚· ★ ⋆｡    ╚════██║   ██║   ██╔══██║██╔══██╗██║     ██╔══██║██╔══██╗╚════██║   ⋆ ✧｡⋆ 
✧ ⋆｡ ˚·    ███████║   ██║   ██║  ██║██║  ██║███████╗██║  ██║██████╔╝███████║   ★ ·˚ ｡
★ ·˚ ✧     ╚══════╝   ╚═╝   ╚═╝  ╚═╝╚═╝  ╚═╝╚══════╝╚═╝  ╚═╝╚═════╝ ╚══════╝   ｡⋆ ✧ 
·˚ ⋆｡⋆｡. ★ ·˚ ★ ·˚ ⋆｡⋆｡. ★ ·˚ ★ ·˚ ⋆｡⋆｡. ★ ·˚ ★ ·˚ ⋆｡⋆｡. ★ ·˚ ⋆｡⋆｡. ★ ·˚ ★ ·˚·˚ ⋆｡⋆｡.
. ⋆ ˚ ✧  . ⋆ ˚  ✦ ˚ . ⋆  ˚ ✦ . ⋆ ˚  ✦ ˚ . ⋆  ˚ ✦ . ⋆ ˚  ✦ ˚ . ⋆  ˚ ✦ . ⋆  ˚ ✦ .. ⋆  ˚ 
✦ ˚ . ⋆   ˚ ✦  ˚  ✦  . ⋆ ˚   ✦  . ⋆ ˚   ✦ ˚ . ⋆   ˚ ✦  ˚  ✦  . ⋆   ˚ ✦  ˚  ✦  . ⋆  ✦"""

    # Создаем градиентный текст
    gradient_logo = Text(logo_text)
    gradient_logo.stylize("bold bright_cyan")

    # Выводим с отступами
    console.print(gradient_logo)
    print()


def show_dev_info():
    """Displays development and version information"""
    console = Console()

    # Создаем красивую таблицу
    table = Table(
        show_header=False,
        box=box.DOUBLE,
        border_style="bright_cyan",
        pad_edge=False,
        width=85,
        highlight=True,
    )

    # Добавляем колонки
    table.add_column("Content", style="bright_cyan", justify="center")

    # Добавляем строки с контактами
    table.add_row("✨ StarLabs Twitter Bot 2.1 ✨")
    table.add_row("─" * 43)
    table.add_row("")
    table.add_row("⚡ GitHub: [link]https://github.com/0xStarLabs[/link]")
    table.add_row("👤 Dev: [link]https://t.me/StarLabsTech[/link]")
    table.add_row("💬 Chat: [link]https://t.me/StarLabsChat[/link]")
    table.add_row(
        "📚 Tutorial: [link]https://star-labs.gitbook.io/star-labs/twitter/eng[/link]"
    )
    table.add_row("")

    # Выводим таблицу с отступом
    print("   ", end="")
    print()
    console.print(table)
    print()



def show_menu(options: List[str]) -> str:
    """
    Shows numbered menu and returns selected option string.
    """
    print("😎  Select Your Option 😎\n")

    # Выводим пронумерованные опции
    for i, option in enumerate(options, 1):
        print(f"[{i}] {option}")

    while True:
        try:
            print("\n")
            choice = input("Your choice: ")
            choices = choice.split(" ")
            options = [options[int(choice) - 1] for choice in choices]
            return options
            
        except ValueError:
            print("     ❌ Please enter a valid number")
